controllers/funding/materials.js

```diff
@@ -317,83 +347,97 @@ function init({ router, routeConfig }) {
         };
     };
 
+    function renderForm(req, res, status = FORM_STATES.NOT_SUBMITTED) {
+        const lang = req.i18n.__(routeConfig.lang);
+        const orders = get(req.session, materialsOrderKey, {});
+        const numOrders = sumBy(values(orders), order => {
+            return order.quantity;
+        });
+
+        res.render(routeConfig.template, {
+            csrfToken: req.csrfToken(),
+            copy: lang,
+            title: lang.title,
+            description: 'Order items free of charge to acknowledge your grant',
+            materials: availableItems,
+            formFields: materialFields,
+            orders: orders,
+            numOrders: numOrders,
+            orderStatus: status
+        });
+    }
+
     /**
      * Free Materials Order Form
      */
     router
-        .route([routeConfig.path])
-        .get(cached.csrfProtection, (req, res) => {
-            let orderStatus;
-
-            // clear order details if it succeeded
-            if (req.flash('materialFormSuccess')) {
-                orderStatus = 'success';
-                delete req.session[materialsOrderKey];
-            } else if (req.flash('materialFormError')) {
-                orderStatus = 'fail';
-            }
-
-            let lang = req.i18n.__(routeConfig.lang);
-            let orders = req.session[materialsOrderKey];
-            let numOrders = 0;
-            if (orders) {
-                for (let o in orders) {
-                    numOrders += orders[o].quantity;
-                }
-            }
-            res.render(routeConfig.template, {
-                title: lang.title,
-                copy: lang,
-                description: 'Order items free of charge to acknowledge your grant',
-                materials: availableItems,
-                formFields: materialFields,
-                orders: orders,
-                numOrders: numOrders,
-                orderStatus: orderStatus,
-                csrfToken: req.csrfToken()
-            });
+        .route(routeConfig.path)
+        .all(cached.csrfProtection)
+        .get((req, res) => {
+            renderForm(req, res, FORM_STATES.NOT_SUBMITTED);
         })
-        .post(validators, cached.csrfProtection, (req, res) => {
-            // sanitise input
-            for (let key in req.body) {
-                req.body[key] = purifyUserInput(req.body[key]);
-            }
+        .post(validators, purify, (req, res) => {
+            const errors = validationResult(req);
 
-            // get form errors and translate them
-            const errors = validationResult(req).formatWith(error => {
```

Comment:
This `formatWith` block has been removed. We no longer need it as we're using the `translateError` stuff as part of the express-validator 5 upgrade.